classes/user/form/ChangePasswordForm.php

```diff
@@ -108,8 +110,15 @@ public function execute(...$functionArgs)
     {
         $user = $this->getUser();
         $user->setPassword(Validation::encryptCredentials($user->getUsername(), $this->getData('password')));
-        parent::execute(...$functionArgs);
+
+        Application::get()->getRequest()->getSessionGuard()->updateUser($user);
+        
+        $user = Auth::logoutOtherDevices($this->getData('password'));
+
         Repo::user()->edit($user);
+
+        parent::execute(...$functionArgs);
```

Comment:
Why is this moved to the end of the function? If it's not necessary, I'd rather avoid it (in case e.g. plugins expect the interaction this way).

Reply:
I think that done mistakenly . and better we keep it as it register a `Hook`. moved it up . before me make changes to session and auth system .